### DISCUSSION TITLE
Align Submenu Height with Parent Menu Height

### DISCUSSION THREAD

**Discussion by spacexengineer** [radix-ui/primitives#1604] [Help]
Hello,

I'm trying to get a submenu to have its height begin/align with the parent menu like the second screenshot below instead of having the submenu align next to the menu item as we see in the current radix menu demo, first screenshot.

I do see that there is a portal container from [the docs](https://www.radix-ui.com/docs/primitives/components/dropdown-menu#:~:text=No%20default%20value-,container,-Prop%20description) but I haven't found a way to get this to align correctly. Ideally, these 2 menus could have the **same parent container div** so that it'd be easier to align the height.

Demo menu as the height starts at the menu item:
![image](https://user-images.githubusercontent.com/6034810/183473700-04ee769a-8288-47f9-9df2-2469afc2547f.png)

Desired height, submenu same as the parent menu:
![image](https://user-images.githubusercontent.com/6034810/183475141-d73d954a-95c6-4cbf-8cf1-0ab45769e9cb.png)

Thanks!

**Comment by spacexengineer** [comment]
ah, I see @jjenzz [mentioned here](https://github.com/radix-ui/primitives/issues/1135#:~:text=The%20container%20prop%20is%20purely%20for%20more%20control%20over%20positioning/styling.) "the `container` prop is purely for more control over positioning/styling".

Is there an item on the roadmap that may enable a solution for handling the styling need here?

**Reply by andy-hook** [reply]
That's correct, the `container` prop is for managing the dom position of portalled elements. You can render the content inline next to the trigger by omitting the `Portal` part. There's no native way to remove the popper positioning but it could be hacked by overriding the rendered styles on `data-radix-popper-content-wrapper` but we don't officially support that.

> Is there an item on the roadmap that may enable a solution for handling the styling need here?

Not at the moment I'm afraid, though we keep an eye on use cases as they are requested.
